test/ValidatorTest.cpp

```diff
@@ -103,19 +157,45 @@ TEST(ValidatorConceptTest, ValidatorConcept) {
       !ValidatorFunction<decltype([](const int&& n) { return n; }), int>);
 
   auto validParameterButFunctionParameterWrongAndReturnTypeWrongTestHelper =
-      []<typename FirstParameter, typename SecondParameter>() {
-        static_assert(
-            !ValidatorFunction<decltype([](FirstParameter n, SecondParameter) {
-                                 return n;
-                               }),
-                               int, int>);
-      };
+      ValidatorConcept_ValidParameterButFunctionParameterWrongAndReturnTypeWrongTestHelper{};
   passCartesianPorductToLambda<
       decltype(validParameterButFunctionParameterWrongAndReturnTypeWrongTestHelper),
       int&, int&&, const int&&>(
       validParameterButFunctionParameterWrongAndReturnTypeWrongTestHelper);
 }
 
+struct
+    ExceptionValidatorConcept_ValidParameterButFunctionParameterWrongAndReturnTypeRightTestHelper {
+  template <typename FirstParameter, typename SecondParameter>
+  void operator()() const {
+    static_assert(!ExceptionValidatorFunction<
+                  decltype([](FirstParameter, SecondParameter) {
+                    return std::optional<ErrorMessage>{};
+                  }),
+                  int, int>);
+  }
+};
+
```

Comment:
In this file also refactor to `type_identity`.

Reply:
Addressed